File: NetBeansProjects/[LFP]Practica/src/Reporte/HTML.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Reporte;

import ListasPractica.ListaTkn;
import ListasPractica.ListaError;
import java.awt.Desktop;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.swing.JOptionPane;
/**
 *
 * @author libni
 */
public class HTML {    
    String Datos = "";
    String ListaB = "";
    ListaTkn Tkn;
    
    public void ReporteToken(List<ListaTkn> arreglo)
    {  
        if(arreglo.size() != 0)
        {
             try
        {            
         FileWriter archivo = new FileWriter("C:/Users/libni/Desktop/ReporteToken.html");
         PrintWriter write = new PrintWriter(archivo);
         write.println("<html>");
         write.println("<head>");
         write.println("<title> LFP PRACTICA NO. 1  TABLA DE TOKEN´S</title>");
         write.println("</head>");
         write.println("<body>");
         write.println("<h1> Listado de Tokens</h1>");
         write.println("<table border>");
         write.println("<tr>");
         write.println("<td><strong>No</strong></td>");
         write.println("<td><strong>Lexema</strong></td>");
         write.println("<td><strong>ID_Token</strong></td>");
         write.println("<td><strong>Token</strong></td>");         
         write.println("</tr>");
         write.println(Datos);
         write.println( "</table>");
         write.println("</body>");
         write.println( "</html>");
        archivo.close();
        Abrir();
        }
        catch(Exception e)
        {
            e.printStackTrace();
        }
        }
        else
        { 
            JOptionPane.showMessageDialog(null, "Upss no hay datos en la lista");            
        }       
    }
    
    public void Abrir()
    {        
     try
     {
         File f = new File("C:/Users/libni/Desktop/ReporteToken.html");
         Desktop.getDesktop().open(f);
     }
     catch(IOException e)
     {
         e.printStackTrace();;
     }
     catch(IllegalArgumentException e)
     {
         JOptionPane.showMessageDialog(null, "Archivo no encontrado", "Error", JOptionPane.ERROR_MESSAGE);
         e.printStackTrace();
     }
    }
    
    public void ReporteError(List<ListaError> datos)
    {
        if(datos.size() != 0)
        {
            for(int i=0; i < datos.size(); i++)
            {
                ListaB = ListaB +"<tr>"
                    + "<td><strong>" + datos.get(i).Enumero + "</strong></td>"
                    + "<td><strong>" + datos.get(i).Elexema + "</strong></td>"                    
                    + "<td><strong>" + datos.get(i).Ecolumna + "</strong></td>"
                    + "<td><strong>" + datos.get(i).Etkn + "</strong></td>"
                    +"</tr>";
            }
        }
        try
        {
         FileWriter archivo = new FileWriter("C:/Users/libni/Desktop/ReporteError.html");
         PrintWriter write = new PrintWriter(archivo);
        write.println("<html>");
        write.println("<head>");
        write.println("<title> LFP PRACTICA NO. 1  TABLA DE ERRORES</title>");
        write.println("</head>");
        write.println("<body>");
        write.println("<h1> Listado de Errores</h1>");
        write.println("<table border>");
        write.println("<tr>");
        write.println("<td><strong>No</strong></td>");
        write.println("<td><strong>Caracter</strong></td>");
        write.println("<td><strong>Fila</strong></td>");
        write.println("<td><strong>Columna</strong></td>");
        write.println("<td><strong>Descripcion</strong></td>");
        write.println("</tr>");
        write.println(ListaB);
        write.println("</table>");
        write.println("</body>");
        write.println("</html>");
        archivo.close();
        EAbrir();
        }
        catch(Exception e)
        {
            e.printStackTrace();
        }
    }
    
    public void EAbrir()
    {        
     try
     {
         File f = new File("C:/Users/libni/Desktop/ReporteError.html");
         Desktop.getDesktop().open(f);
     }
     catch(IOException e)
     {
         e.printStackTrace();;
     }
     catch(IllegalArgumentException e)
     {
         JOptionPane.showMessageDialog(null, "Archivo no encontrado", "Error", JOptionPane.ERROR_MESSAGE);
         e.printStackTrace();
     }
    }
}
/*
for(int i=0; i < arreglo.size(); i++)
            {                   
                Datos = Datos + "<tr>" 
                    + "<td><strong>" + arreglo.get(i).getNumero()  + "</strong></td>"
                    + "<td><strong>" + arreglo.get(i).getLexema() + "</strong></td>"
                    + "<td><strong>" + arreglo.get(i).getIdtkn() + "</strong></td>"
                    + "<td><strong>" + arreglo.get(i).getTkn() + "</strong></td>"
                    + "<td><strong>" + arreglo.get(i).getFila() + "</strong></td>"
                    + "<td><strong>" + arreglo.get(i).getColumna() + "</strong></td>"                    
                    + "</tr>" + "\n";
                System.out.println("numero: " +arreglo.get(i).getNumero()+",lexema: " +arreglo.get(i).getLexema() );
            }  
*/

File: LFP_PROYECTO_1/[LFP]Proyecto1_201403541/Html.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _LFP_Proyecto1_201403541
{
    class Html
    {
        string tabla = "";
        string Etabla = "";

        public void ReporteTKN(List<Listas> datos)
        {
            if (datos.Count != 0)
            {
                for (int i = 0; i < datos.Count; i++)
                {
                    tabla = tabla + "<tr>"
                      + "<td><strong>" + datos[i].numero + "</strong></td>"
                      + "<td><strong>" + datos[i].lexema + "</strong></td>"
                      + "<td><strong>" + datos[i].idtkn + "</strong></td>"
                      + "<td><strong>" + datos[i].tkn + "</strong></td>"
                      + "<td><strong>" + datos[i].fila + "</strong></td>"
                      + "<td><strong>" + datos[i].columna + "</strong></td>"
                      + "</tr>";
                }
            }

            string[] texto = { "<html>"
                    ,"<head>"
                ,"<title>TABLA DE TOKEN'S</title>"
                ,"</head>"
                ,"<body>"
                ,"<h1> LFP PROYECTO NO.1 Listado de Tokens</h1>"
                ,"<table border>"
                ,"<tr>"
                ,"<td><strong>No</strong></td>"
                ,"<td><strong>Lexema</strong></td>"
                ,"<td><strong>ID_Token</strong></td>"
                ,"<td><strong>Token</strong></td>"
                ,"<td><strong>Fila</strong></td>"
                ,"<td><strong>Coluna</strong></td>"
                ,"</tr>"
                ,tabla
                ,"</table>"
                ,"</body>"
                ,"</html> "
                };
            System.IO.File.WriteAllLines(@"C:\Users\libni\OneDrive\Escritorio\ReporteToken.html", texto);
        }

        public void ReporteETKN(List<Elista> valor)
        {
            if (valor.Count != 0)
            {
                for (int i = 0; i < valor.Count; i++)
                {
                    Etabla = Etabla + "<tr>"
                        + "<td><strong>" + valor[i].num + "</strong></td>"
                        + "<td><strong>" + valor[i].fil + "</strong></td>"
                        + "<td><strong>" + valor[i].col + "</strong></td>"
                        + "<td><strong>" + valor[i].lex + "</strong></td>"
                        + "<td><strong>" + valor[i].Etkn + "</strong></td>"
                        + "</tr>";
                }
            }

            string[] text = { "<html>"
                ,"<head>"
                ,"<title>TABLA DE ERRORES</title>"
                ,"</head>"
                ,"<body>"
                ,"<h1> LFP PROYECTO NO.1 Listado de Errores</h1>"
                ,"<table border>"
                ,"<tr>"
                ,"<td><strong>No</strong></td>"
                ,"<td><strong>Fila</strong></td>"
                ,"<td><strong>Columna</strong></td>"
                ,"<td><strong>Caracter</strong></td>"
                ,"<td><strong>Descripcion</strong></td>"
                ,"</tr>"
                ,Etabla
                ,"</table>"
                ,"</body>"
                ,"</html> " };
            System.IO.File.WriteAllLines(@"C:\Users\libni\OneDrive\Escritorio\ReporteError.html", text);
        }
    }
}
